junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ExtensionRegistry.java

```diff
@@ -88,28 +90,47 @@ private ExtensionRegistry(ExtensionRegistry parent) {
 	}
 
 	/**
-	 * Get all {@code Extensions} in this registry or one of its ancestors of
+	 * Stream all {@code Extensions} in this registry or one of its ancestors of
 	 * the specified type.
 	 *
 	 * @param extensionType the type of {@link Extension} to get
 	 * @see #getReversedExtensions(Class)
-	 * @see #stream(Class)
+	 * @see #getExtensions(Class)
 	 */
-	public <E extends Extension> List<E> getExtensions(Class<E> extensionType) {
-		List<E> extensions = new ArrayList<>();
-
-		if (this.parent != null) {
-			extensions.addAll(this.parent.getExtensions(extensionType));
+	public <E extends Extension> Stream<E> streamExtensions(Class<E> extensionType) {
```

Comment:
Hi guys,

While I certainly appreciate your attention to detail, let's not go overboard here. This is _just_ an internal API. 😉 

The reason the method is currently called `stream` is for DSL-like conciseness. 

Although I can live with `streamExtensions`, I will personally veto `getExtensionsAsStream` since there is no need to be overly verbose here.

The _thing_ in question is an extension registry. Thus getting a stream from it will almost certainly _stream_ the extensions contained in that registry. Honestly, what else would you expect such a method to do? Plus, the method clearly documents what it does.

In summary, I welcome the other changes in this PR, but I don't see a concrete need for renaming the `stream` method.

@junit-team/junit-lambda, what do the rest of you think?